src/context/Web3Context.tsx

```diff
@@ -260,6 +264,14 @@ const Web3Provider = ({
           );
         }
 
+        if (token.attachFunctions) {
+          newTokenInfo.functions = {
+            allowance: tokenContract.allowance,
```

Comment:
These functions should be wrapped in a `useCallback()` and exposed directly on the the `TokenInfo` object on line 234 
```
        const newTokenInfo: TokenInfo = {
          decimals: 0,
          balance: 0,
          name: token.name,
          symbol: token.symbol,
          imageUri: token.imageUri,
          allowance: useCallback()
          ....
        };
```